host/sgx/windows/debugrtbridge.c

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <Windows.h>
+#include <openenclave/internal/debugrt/host.h>
+#include <openenclave/internal/trace.h>
+#include <stdlib.h>
+#include "../../hostthread.h"
+
+static struct
+{
+    oe_once_type once;
+    HMODULE hmodule;
+    oe_result_t (*notify_enclave_created)(oe_debug_enclave_t* enclave);
+    oe_result_t (*notify_enclave_terminated)(oe_debug_enclave_t* enclave);
+    oe_result_t (*push_thread_binding)(
+        oe_debug_enclave_t* enclave,
+        struct _sgx_tcs* tcs);
+    oe_result_t (*pop_thread_binding)(void);
+} _oedebugrt;
+
+static void get_debugrt_function(const char* name, FARPROC* out)
+{
+    *out = GetProcAddress(_oedebugrt.hmodule, name);
+    if (*out == NULL)
+    {
+        OE_TRACE_FATAL("Could not find function %s in oedebugrt.dll", name);
+    }
+}
+
+static void load_oedebugrt(void)
+{
+    if (_oedebugrt.hmodule != NULL)
+    {
+        OE_TRACE_WARNING("oedebugrt.dll has already been loaded.");
+        return;
+    }
+
+#ifndef NDEBUG
+    /**
+     *  In debug mode, give preference to OE_DEBUGRT_PATH.
+     *  This is mainly used for OE SDK development.
+     */
+    {
+        char* debugrtpath = getenv("OE_DEBUGRT_PATH");
+        _oedebugrt.hmodule = LoadLibraryExA(
+            debugrtpath,
+            NULL, /* reserved */
+            /* Search only specified path. */
+            LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
+    }
+
+#endif
+
+    /* Search for oedebugrt.dll only in the application folder. */
+    if (_oedebugrt.hmodule == NULL)
+    {
+        _oedebugrt.hmodule = LoadLibraryExA(
+            "oedebugrt.dll",
+            NULL, /* reserved */
+            LOAD_LIBRARY_SEARCH_APPLICATION_DIR);
```

Comment:
Sorry, last question here. The way I normally stored my debugger extensions for windbg was under c:\debuggers (where windbg was installed). Using LOAD_LIBRARY_SEARCH_APPLICATION_DIR  would need app developers to keep copies of the extension with every app. Is there another way to do it? I dont know what the best practices from the debugging team are. If this is what is recommended, please ignore. Would using "LOAD_LIBRARY_SEARCH_DEFAULT_DIRS" be better?

Reply:
Just to be clear. This is oedebugrt.dll is **not** the debugger extension. It is part of the SDK runtime.

@CodeMonkeyLeet  has a better idea of how the OE DLLs (say for attestation, enclave platform, oedebugrt etc) will be distributed. When that is determined, we will have to change the way we search the debug rt dll.

Reply:
Thanks for the explanation. I thought it was a debugger extension.